cli/src/main/java/org/jboss/as/cli/operation/impl/DefaultOperationCandidatesProvider.java

```diff
@@ -336,6 +310,73 @@ public CommandLineCompleter getValueCompleter() {
         return result;
     }
 
+    private CommandLineCompleter getCompleter(final Map<String, CommandLineCompleterFactory> globalOpProps, final Property prop, CommandContext ctx, String operationName, OperationRequestAddress address) throws IllegalArgumentException {
+        CommandLineCompleter propCompleter = null;
+        final CommandLineCompleterFactory factory = globalOpProps == null ? null : globalOpProps.get(prop.getName());
+        if (factory != null) {
+            final String propName = ctx.getParsedCommandLine().getPropertyValue(Util.NAME);
+            if (Util.WRITE_ATTRIBUTE.equals(operationName) && propName != null) {
```

Comment:
We could have the FilenameTabCompleter.translatePath take the CommandContext as argument, thus we could have a constructor without CommandContext, and some changes in FileSystemPathArgument which would then requires the CommandContext in its constructor.